### PR TITLE
I2C Light Sensor

### PR DESCRIPTION
**Pull request by aselafernando**
Added a plugin to set night mode based on the value read from an i2c light sensor. Different sensors can be added through json files in the sensor folder.
Reading of the sensor can be disabled by setting the refresh interval to 0 (default).

It'll read the ambient light level from the specified register in the json file. In addition it has the ability to write a value to a configuration register each time, in case the module needs a write to turn it on.

This requires libi2c-dev be installed to compile (will need the wiki updated).

### CODE REVIEW

**Comment by viktorgino** [comment]
Thanks @aselafernando! I'll have a proper look at it this weekend

File: modules/i2c-light-sensor/i2c-light-sensor-plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,251 @@
+#include "i2c-light-sensor-plugin.h"
+
+I2CLightSensorPlugin::I2CLightSensorPlugin(QObject *parent) : QObject(parent), m_refreshTimer(this)
+{
+    m_pluginSettings.eventListeners = QStringList() << "AndroidAuto::connected";
+    //m_pluginSettings.events = QStringList() << "Brightness";
+    //m_pluginSettings.actions = QStringList() << "GetBrightness";
```
Comment (viktorgino) [inline modules/i2c-light-sensor/i2c-light-sensor-plugin.cpp]: Could you remove these please?

File: modules/i2c-light-sensor/i2c-light-sensor-plugin.cpp [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        stopTimer();
+    }
+}
+
+void I2CLightSensorPlugin::stopTimer() {
+    m_refreshTimer.stop();
+}
+
+void I2CLightSensorPlugin::eventMessage(QString id, QVariant message) {
+    //qDebug() << "I2CLightSensor: event from " << id;
```
Comment (viktorgino) [inline modules/i2c-light-sensor/i2c-light-sensor-plugin.cpp]: Remove?

File: modules/i2c-light-sensor/i2c-light-sensor-plugin.cpp [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    //m_pluginSettings.actions = QStringList() << "GetBrightness";
+}
+
+QObject *I2CLightSensorPlugin::getContextProperty() {
+    return this;
+}
+
+void I2CLightSensorPlugin::PrintString(char *message, int length) {
+    if (length > 0) {
+        qDebug() << "I2CLightSensor: " << message;
```
Comment (viktorgino) [inline modules/i2c-light-sensor/i2c-light-sensor-plugin.cpp]: Maybe use qCDebug?

**Review by viktorgino — CHANGES_REQUESTED** [review]
Thanks @aselafernando , just a few changes and good to go.
FYI back to working on this project again :)

**Comment by aselafernando** [comment]
Made the changes, I also corrected a bug in the pro file where it wasn't copying the sensor json files over to the install directory.

I left ```m_pluginSettings.eventListeners = QStringList() << "AndroidAuto::connected";``` in as the plugin does rely on this event to ensure that as soon as a phone is connected it sets the night mode value.

**Comment by viktorgino** [comment]
> Made the changes, I also corrected a bug in the pro file where it wasn't copying the sensor json files over to the install directory.
> 
> I left `m_pluginSettings.eventListeners = QStringList() << "AndroidAuto::connected";` in as the plugin does rely on this event to ensure that as soon as a phone is connected it sets the night mode value.

Perfect, thanks!

**Pull request merged**